rmw_fastrtps_shared_cpp/src/rmw_node.cpp

```diff
@@ -42,45 +47,60 @@ __rmw_create_node(
   const char * name,
   const char * namespace_)
 {
-  if (!name) {
-    RMW_SET_ERROR_MSG("name is null");
+  assert(nullptr != context);
+  assert(nullptr != context->impl);
+  assert(identifier == context->implementation_identifier);
```

Comment:
I'd use the `RMW_CHECK_ARGUMENT_FOR_NULL` macros here.

Reply:
We could, though we'd be replicating the checks that occur in the actual RMW implementations. Since this is a somewhat "internal" API (not a standalone RMW implementation but the pieces of it), I figured we may as well keep these as preconditions instead of checking twice.

Reply:
Good point about it being an internal API.  With that in mind, I think we can elide the checks completely.  The basic idea here being that a library should be able to trust itself; anything else is just buggy.  And as you say, these are duplicate checks anyway.

(the same goes for `__rmw_destroy_node` below)

Reply:
Agreed. I'd keep the implementation identifier assertion though, since it is a precondition we can fully enforce (as opposed to pointer validity) and failing to meet it implies an ill-formed program (that we want to catch as soon as possible if not preclude entirely).